client/src/components/Form/components/FormSubmit.tsx

```diff
@@ -20,83 +20,84 @@ interface FormSubmitProps {
-                    })
-                        .then((res) => {
-                            if (res.status === 204) {
-                                window.frontendlogger.info(`Oppgave med oppgaveid: ${oppgaveid} ble registrert`);
-                                setSuccessModalContent('Oppgaven ble ferdigstilt.');
-                            } else if (res.status === 400) {
-                                // Try to parse body as json. if it fails, catch the error and parse the body as text instead.
-                                // Clone because the res.body stream cannot be read more than one time.
-                                return res
-                                    .clone()
-                                    .json()
-                                    .then((json) => {
-                                        return iotsPromise.decode(RuleHitErrors, json);
-                                    })
-                                    .then((ruleHitErrors) => {
-                                        setRuleHitErrors(ruleHitErrors);
-                                    })
-                                    .catch((error) => {
-                                        // If the json received is not of the expected shape
-                                        if (iotsPromise.isDecodeError(error)) {
-                                            throw new Error('Det oppsto en valideringsfeil. Feilkode: ' + res.status);
-                                        } else {
-                                            return res.text();
-                                        }
-                                    });
-                            } else if ([401, 404, 500].includes(res.status)) {
-                                return res.text();
-                            } else {
-                                throw new Error('Det oppsto en feil i baksystemet med feilkode: ' + res.status);
-                            }
-                        })
-                        .then((errorText) => {
-                            if (errorText) {
-                                throw new Error(errorText);
-                            }
-                        })
-                        .catch((error) => {
-                            setSuccessError(error);
-                            window.frontendlogger.error(error);
-                        })
-                        .finally(() => setIsLoadingSuccess(false));
+        setError(null);
+        setRuleHitError(null);
+
+        if (!enhet) {
+            setError(new Error('Enhet mangler. Vennligst velg enhet fra nedtrekksmenyen øverst på siden'));
+            return;
+        }
+
+        handleSubmit(async (formState) => {
+            const sykmelding = buildRegistrertSykmelding(formState);
+
+            if (!sykmelding) {
+                window.frontendlogger.error('Noe gikk galt med konstruksjon av sykmeldingsobjekt');
+                return;
+            }
```

Comment:
Er dette hovedsaklig for å tilfredsstille TypeScript, aka "det skal aldri skje"? For dersom noen av disse treffes så blir det vel ikke oppdatert noe feedback til brukeren at noe har gått galt?

Reply:
Del av et større problem som burde ryddes opp i... APIet tar ikke imot dataen slik den ser ut i skjemaet. Den tar det imot på et annet format som passer med slik den ser ut i databasen. Så her forsøkes det å transformere skjemaet. Hvis noen av feltene på en eller annen måte passerte valideringen som er for skjemaet, så vil det være en ekstra sjekk for om påkrevde felter faktisk er der, og hvis ikke så returneres undefined.

Mitt ønske er at APIet tar imot skjemadataen som den er, og gjør nødvendig transformering og validering, og så sier fra hvis noe er feil. Man håper jo at skjemavalideringen i frontend er riktig først og fremst. Hva tenker du?

Reply:
Kan uansett gi en feilmelding til bruker. det er nok lurt